### PR TITLE
Make extract functions signatures available for gpu build

### PR DESCRIPTION
**Pull request by kurapov-peter**
These were undefined for gpu build.

### CODE REVIEW

File: Tests/ArrowBasedExecuteTest.cpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ const size_t g_num_rows{10};
 
 class ExecuteTestBase {
  public:
-  static bool gpusPresent() { return gpusPresent(); }
```
Comment (kurapov-peter) [inline Tests/ArrowBasedExecuteTest.cpp:84]: Didn't even notice when removed the namespace :)

File: Tests/ArrowBasedExecuteTest.cpp [hunk 2]
```diff
@@ -81,7 +81,7 @@ const size_t g_num_rows{10};
...
+  static bool hasPresent() { return gpusPresent(); }
```
Comment (ienkovich) [inline Tests/ArrowBasedExecuteTest.cpp:84]:
```suggestion
```

File: Tests/ArrowBasedExecuteTest.cpp [hunk 3]
```diff
@@ -1570,7 +1570,7 @@ class ExecuteTestBase {
 
 bool skip_tests(const ExecutorDeviceType device_type) {
 #ifdef HAVE_CUDA
-  return device_type == ExecutorDeviceType::GPU && !(ExecuteTestBase::gpusPresent());
+  return device_type == ExecutorDeviceType::GPU && !(ExecuteTestBase::hasPresent());
```
Comment (ienkovich) [inline Tests/ArrowBasedExecuteTest.cpp:1573]:
```suggestion
  return device_type == ExecutorDeviceType::GPU && !(gpusPresent());
```

**Review by ienkovich — COMMENTED** [review]
`hasPresent` is misleading :)

**Comment by kurapov-peter** [comment]
> `hasPresent` is misleading :)

The function's cursed! :)

**Comment by alexbaden** [comment]
I had to add this patch to get it to build on clang-13:

```
diff --git a/QueryEngine/Descriptors/QueryFragmentDescriptor.cpp b/QueryEngine/Descriptors/QueryFragmentDescriptor.cpp
index 4e38b6f49..a94e35d13 100644
--- a/QueryEngine/Descriptors/QueryFragmentDescriptor.cpp
+++ b/QueryEngine/Descriptors/QueryFragmentDescriptor.cpp
@@ -152,7 +152,7 @@ void QueryFragmentDescriptor::buildFragmentPerKernelForTable(
     }
 
     ExecutionKernelDescriptor execution_kernel_desc{
-        device_id, {}, get_fragment_tuple_count(fragment)};
+        static_cast<int>(device_id), {}, get_fragment_tuple_count(fragment)};
     if (table_desc_offset) {
       const auto frag_ids =
           executor->getTableFragmentIndices(ra_exe_unit,
@@ -367,7 +367,7 @@ void QueryFragmentDescriptor::buildMultifragKernelMap(
       if (execution_kernels_per_device_[device_type].find(device_id) ==
           execution_kernels_per_device_[device_type].end()) {
         std::vector<ExecutionKernelDescriptor> kernel_descs{
-            ExecutionKernelDescriptor{device_id, FragmentsList{}, std::nullopt}};
+            ExecutionKernelDescriptor{static_cast<int>(device_id), FragmentsList{}, std::nullopt}};
         CHECK(execution_kernels_per_device_[device_type]
                   .insert(std::make_pair(device_id, kernel_descs))
                   .second);
~
```

**Comment by kurapov-peter** [comment]
> I had to add this patch to get it to build on clang-13:
> 
> ```
> diff --git a/QueryEngine/Descriptors/QueryFragmentDescriptor.cpp b/QueryEngine/Descriptors/QueryFragmentDescriptor.cpp
> index 4e38b6f49..a94e35d13 100644
> --- a/QueryEngine/Descriptors/QueryFragmentDescriptor.cpp
> +++ b/QueryEngine/Descriptors/QueryFragmentDescriptor.cpp
> @@ -152,7 +152,7 @@ void QueryFragmentDescriptor::buildFragmentPerKernelForTable(
>      }
>  
>      ExecutionKernelDescriptor execution_kernel_desc{
> -        device_id, {}, get_fragment_tuple_count(fragment)};
> +        static_cast<int>(device_id), {}, get_fragment_tuple_count(fragment)};
>      if (table_desc_offset) {
>        const auto frag_ids =
>            executor->getTableFragmentIndices(ra_exe_unit,
> @@ -367,7 +367,7 @@ void QueryFragmentDescriptor::buildMultifragKernelMap(
>        if (execution_kernels_per_device_[device_type].find(device_id) ==
>            execution_kernels_per_device_[device_type].end()) {
>          std::vector<ExecutionKernelDescriptor> kernel_descs{
> -            ExecutionKernelDescriptor{device_id, FragmentsList{}, std::nullopt}};
> +            ExecutionKernelDescriptor{static_cast<int>(device_id), FragmentsList{}, std::nullopt}};
>          CHECK(execution_kernels_per_device_[device_type]
>                    .insert(std::make_pair(device_id, kernel_descs))
>                    .second);
> ~
> ```

I could make it all ints, just wanted to start moving to unsigned. I guess, that's not a good way of doing it anyway.

**Comment by alexbaden** [comment]
I would go with all ints for now. We could make it a type... `using DeviceNum = int` or some such.

**Review by ienkovich — APPROVED** [review]

**Review by alexbaden — APPROVED** [review]

**Pull request merged**